pkg/kubeapiserver/resource_handler.go

```diff
@@ -73,17 +86,7 @@ func (r *ResourceHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	// Check the health of the cluster
 	if clusterName != "" {
-		cluster, err := r.clusterLister.Get(clusterName)
-		if err != nil {
-			err := fmt.Errorf("not found request cluster")
-			klog.ErrorS(err, "Failed to handle resource request, not get cluster from cache", "cluster", clusterName, "resource", gvr)
-			responsewriters.ErrorNegotiated(
-				apierrors.NewInternalError(err),
-				Codecs, gvr.GroupVersion(), w, req,
-			)
-			return
-		}
-
+		cluster, _ := r.clusterLister.Get(clusterName)
```

Comment:
It is possible to use the cluster got in the previous step, because we can't guarantee that getting it from Lister again will return the cluster

Reply:
done